test/units/testsuite-82.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+set -ex
+set -o pipefail
+
+systemd-analyze log-level debug
+
+export SYSTEMD_LOG_LEVEL=debug
+
+if [ -f /run/testsuite82.touch ] ; then
+    # This is the second boot
+    systemd-notify --status="Second Boot"
+
+    # Clean up what we created earlier
+    rm /run/testsuite82.touch
+
+    # Check that the fdstore entry still exists
+    test "$LISTEN_FDS" -eq 1
+    read -r x <&3
+    test "$x" = "wuffwuff"
+else
+    # This is the first boot
+    systemd-notify --status="First Boot"
+
+    # Let's upload an fd to the fdstore, so that we can verify fdstore passing works correcly
+    T="/dev/shm/fdstore.$RANDOM"
+    echo "wuffwuff" > "$T"
+    systemd-notify --fd=3 --pid=parent 3<"$T"
+    rm "$T"
+
```

Comment:
would be good to beef up this test, to add a normal service that is shut down as expected (and verify that it is), and one that survives (and verify that it does)